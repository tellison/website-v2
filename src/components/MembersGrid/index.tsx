import * as React from 'react'

const MembersGrid = (members) => {

const array = members.members

    return (
        <div className='container'>
            <div className='row justify-content-center align-items-center'>
                {array.map(
                    (member, i): string | JSX.Element =>
                        member && (
                            <div className='col-lg-4 col-md-12 mt-5' key={member.name}>
                                <a target='_blank' rel='noopener noreferrer' href={member.url}>
                                    <img src={`/images/${member.logo}`} style={ member.logoPadding ? { paddingRight: member.logoPadding, paddingLeft: member.logoPadding, width: '10rem'} : {width: '10rem'}} alt={`${member.name} logo`}/>
                                </a>
                            </div>
                    )
                )}
            </div>
        </div>
    );
};

export default MembersGrid;