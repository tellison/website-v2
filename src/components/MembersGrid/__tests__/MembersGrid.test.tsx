import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import MembersGrid from '..';

describe('MembersGrid component', () => {

  const members = [
    {
        name: 'Member 1',
        logo: 'member1.png',
        url: 'https://member1.com'
    },
    {
        name: 'Member 2',
        logo: 'member2.png',
        logoPadding: '1em',
        url: 'https://member2.com'
    }
  ]

  it('Members grid renders correctly', () => {
    const { container } = render(
      <MembersGrid members={members}/>
    );
    expect(container).toMatchSnapshot();
  });

});