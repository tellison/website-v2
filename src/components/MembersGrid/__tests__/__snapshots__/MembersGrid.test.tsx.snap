// Vitest Snapshot v1

exports[`MembersGrid component > Members grid renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row justify-content-center align-items-center"
    >
      <div
        class="col-lg-4 col-md-12 mt-5"
      >
        <a
          href="https://member1.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Member 1 logo"
            src="/images/member1.png"
            style="width: 10rem;"
          />
        </a>
      </div>
      <div
        class="col-lg-4 col-md-12 mt-5"
      >
        <a
          href="https://member2.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Member 2 logo"
            src="/images/member2.png"
            style="padding-right: 1em; padding-left: 1em; width: 10rem;"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
